### PR TITLE
feat: (Platform) introduce Time Picker component

### PR DESCRIPTION
**Pull request by EvilAlexei**
#### Please provide a link to the associated issue.
No issue
#### Please provide a brief summary of this pull request.
This PR contains a platform component that wraps core Time Picker to work with Platform Forms.
#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/platform/src/lib/components/form/time-picker/time-picker.component.spec.ts
Comment (Betrozov) [inline libs/platform/src/lib/components/form/time-picker/time-picker.component.spec.ts:5]: please add tests

**Review by Betrozov — CHANGES_REQUESTED** [review]

File: libs/platform/src/lib/components/form/time-picker/time-picker.component.spec.ts
Comment (EvilAlexei) [reply libs/platform/src/lib/components/form/time-picker/time-picker.component.spec.ts]: @Betrozov Please re-check :)

File: libs/platform/src/lib/components/form/time-picker/time-picker.component.spec.ts
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/form/time-picker/time-picker.component.spec.ts:59]: please remove the focus

File: libs/platform/src/lib/components/form/time-picker/time-picker.component.ts
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/form/time-picker/time-picker.component.ts:159]: If you want you can change this to ternary expression to make it more clear:
```
!this.timePickerComponent.time && !this.allowNull ? this.state = 'error' : this.state = undefined
```

File: libs/platform/src/lib/components/form/time-picker/time-picker.component.ts
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/form/time-picker/time-picker.component.ts:168]: I think it should be named `_getPlaceholder()` as it's hidden (as discussed with your team)

File: libs/platform/src/lib/components/form/time-picker/time-picker.component.ts
Comment (EvilAlexei) [reply libs/platform/src/lib/components/form/time-picker/time-picker.component.ts:159]: @InnaAtanasova for me it's more readable that way :) but not a big deal, I can change it if you think it will be better.

File: libs/platform/src/lib/components/form/time-picker/time-picker.component.ts
Comment (EvilAlexei) [reply libs/platform/src/lib/components/form/time-picker/time-picker.component.ts:168]: I'm not a fan of this decision, but it's already discussed and approved )))))) I will change it

**Review by Betrozov — APPROVED** [review]

File: apps/docs/src/app/platform/component-docs/platform-forms/platform-time-picker/platform-time-picker-docs.component.html
Comment (salarenko) [inline apps/docs/src/app/platform/component-docs/platform-forms/platform-time-picker/platform-time-picker-docs.component.html:15]: Use kebab-case for id attributes. `componentName` value needs to match component sub-page url (time-picker)

File: libs/platform/src/lib/components/form/time-picker/time-picker.component.ts
Comment (salarenko) [inline libs/platform/src/lib/components/form/time-picker/time-picker.component.ts:76]: Be consistent. Use `@Input` docs annotation for other inputs.

**Review by salarenko — CHANGES_REQUESTED** [review]
- There is probably an empty popover body opening when time-picker in reactive forms  is hovered
time-picker-empty-popover

File: apps/docs/src/app/platform/component-docs/platform-forms/platform-time-picker/platform-time-picker-docs.component.html
Comment (EvilAlexei) [reply apps/docs/src/app/platform/component-docs/platform-forms/platform-time-picker/platform-time-picker-docs.component.html:15]: @salarenko sorry, could you please clarify about `componentName` value? What do you mean?

I tried to find a pattern in the project for this, but there are a lot of camelCase and kebab-case in such cases, so
I will be glad on clarifying the rules for that.

**Comment by EvilAlexei** [comment]
@salarenko I was able to reproduce an empty popover body, but this is a common issue for 
`<fdp-form-field>`. 

For example:
image

Are you ok If I create a separate issue for that?

**Comment by EvilAlexei** [comment]
@salarenko after I merged the latest master I can't reproduce this issue, with an empty popover body, I think It was fixed.

File: libs/platform/src/lib/components/form/time-picker/time-picker.component.ts
Comment (InnaAtanasova) [reply libs/platform/src/lib/components/form/time-picker/time-picker.component.ts]: my message starts with `If you want...`. I never insist on other people writing code the way I do :)

Comment (InnaAtanasova) [reply libs/platform/src/lib/components/form/time-picker/time-picker.component.ts]: And I didn't request changes, meaning all my comments are "debatable". If I think something must be fix it will be evident :P (just for future reference)

**Review by InnaAtanasova — APPROVED** [review]

File: libs/platform/src/lib/components/form/time-picker/time-picker.component.ts
Comment (InnaAtanasova) [reply libs/platform/src/lib/components/form/time-picker/time-picker.component.ts]: I didn't approve because there was no approval from Platform team and our agreement is to not approve Platform PRs without at least 1 approval from your team.

**Review by salarenko — APPROVED** [review]

**Comment by dimamarksman** [comment]
Let's postpone this merge until https://github.com/SAP/fundamental-ngx/pull/3920 is done (changes introduced in #3920 will infulnce changes in this PR)

File: libs/platform/src/lib/components/form/time-picker/time-picker.component.html
Comment (dimamarksman) [inline libs/platform/src/lib/components/form/time-picker/time-picker.component.html:23]: I'm wondering if it's valid to keep `[time]="time"` model along with the `[(ngModel)]="value"`.
What is the source of truth in this case?

Comment (EvilAlexei) [reply libs/platform/src/lib/components/form/time-picker/time-picker.component.html:23]: @dimamarksman, I don't know about a strong source of truth, I saw a similar implementation in platform datepicker.

In my opinion, it's a good way to provide all available options to manage the control, and it's a common practice to have an `ngModel` on the "input"/form element.

In any case, I think it's better to discuss this point with the platform team directly.

Comment (dimamarksman) [reply libs/platform/src/lib/components/form/time-picker/time-picker.component.html:23]: I meant that we probably need to control underlying core-time-picker using only one way - or [ngModel] or [time]

Comment (dimamarksman) [reply libs/platform/src/lib/components/form/time-picker/time-picker.component.html:23]: Probably not a big deal, just found it a little bit confusing

**Comment by Betrozov** [comment]
@EvilAlexei please resolve conflicts

**Review by JKMarkowski — APPROVED** [review]
Works well good for me

File: apps/docs/src/app/platform/component-docs/platform-forms/platform-time-picker/examples/platform-time-picker-template-example.component.ts
Comment (dimamarksman) [inline apps/docs/src/app/platform/component-docs/platform-forms/platform-time-picker/examples/platform-time-picker-template-example.component.ts:11]: Since new `DatetimeAdapter` has been introduced there is no `TimeObject` anymore instead we should use FdDatetime instace (it's common for time and date components)

**Review by dimamarksman — CHANGES_REQUESTED** [review]
LGTM
Looks like it's not rebased

**Comment by mikerodonnell89** [comment]
@EvilAlexei build is failing:

```
7:26:12 AM: Error: /opt/build/repo/libs/platform/src/lib/components/form/time-picker/time-picker.component.ts(14)
7:26:12 AM:  Module '"../../../../../../core/src"' has no exported member 'TimeObject'.
7:26:12 AM: Error: /opt/build/repo/libs/platform/src/lib/components/form/time-picker/time-picker.component.ts(129)
7:26:12 AM:  Generic type 'TimePickerComponent<D>' requires 1 type argument(s).
```

**Review by salarenko — APPROVED** [review]

**Review by dimamarksman — APPROVED** [review]

**Comment by EvilAlexei** [comment]
Closed for #4212

**Pull request closed without merging**
